src/cobalt.rs

```diff
@@ -131,9 +131,19 @@ pub fn build(config: &Config) -> Result<()> {
         .collect();
 
     trace!("Generating posts");
-    for mut post in &mut posts {
+    for (i, mut post) in &mut posts.iter_mut().enumerate() {
         trace!("Generating {}", post.path);
 
+        // posts are in reverse date order, so previous post is the next in the list (+1)
+        if let Some(previous) = simple_posts_data.get(i + 1) {
+            post.attributes.insert("previous".to_owned(), previous.clone());
+        }
+        if i >= 1 {
+            if let Some(next) = simple_posts_data.get(i - 1) {
```

Comment:
I had memory and processing performance concerns but in thinking about them, they are a non-issue.

I'd like to make cobalt multi-threaded in the future.  If we view this all as stages in a pipeline, we can make each stage threaded but we won't make things threaded across stages.

I also recoil a little at the amount of memory allocated.  I considered us passing along only specific parts of `document.attributes` but we can't predict too well what the user might want.

So should we make this optional? The problem is most people probably want this.

So in the end, oh well.